extensions/wikia/Wall/WallThread.class.php

```diff
@@ -26,11 +31,33 @@ protected function initializeReplyData() {
 	 * @param int $id
 	 * @return WallThread
 	 */
-	static public function newFromId( $id ) {
-		$wt = new WallThread();
-		$wt->mThreadId = $id;
+	static public function newFromId( int $id ) : WallThread {
+		return new WallThread( $id );
+	}
+
+	/**
+	 * @param int[] $ids
+	 * @return WallThread[]
+	 */
+	static public function newFromIds( array $ids ) : array {
+		$threads = [];
+
+		// fetch main wall messages for requested threads in a single batch
+		$wall_messages = WallMessage::newFromIds( $ids );
+
+		foreach( $ids as $i => $id ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$i" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)